File: C++/shuffle.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
        vector<int> shuffle(vector<int>& nums, int n) {
            vector<int> result(2*n);
            // printf("result.size() = %d\n", result.size());
            for(int i = 0; i < n; i++){
                result[2 * i] = nums[i];
                result[2 * i + 1] = nums[i + n];
            }
            return result;
        }
    };

int main(){
    Solution solution;
    vector<int> nums = {2, 5, 1, 3, 4, 7};
    vector<int> result = solution.shuffle(nums, 3);
    for(int i = 0; i < result.size(); i++){
        cout << result[i] << " ";
    }
    return 0;
}

File: Newbie_basics/Python/shuffle.py
from typing import List

class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        result = []
        for i in range(n):
            result.append(nums[i])
            result.append(nums[i + n])
        return result

def main():
    solution = Solution()
    print(solution.shuffle([2, 5, 1, 3, 4, 7], 3))

if __name__ == "__main__":
    main()

  
    


